src/ngraph/op/one_hot.cpp

```diff
@@ -34,6 +34,10 @@ void op::OneHot::validate_and_infer_types()
     PartialShape arg_shape = get_input_partial_shape(0);
     Rank arg_rank = arg_shape.rank();
 
+    NODE_VALIDATION_CHECK(this,
+                          arg_et.is_dynamic() || !arg_et.is_real(),
```

Comment:
`is_integral`

Reply:
What do you mean by `is_integral` here? It's not something that is defined yet in `element::Type`.